src/components/DependencyTree/DependencyTree.tsx

```diff
@@ -0,0 +1,262 @@
+import { Button, Divider, TreeViewDataItem } from '@patternfly/react-core';
+import { TreeView } from '@patternfly/react-core';
+import { Build, GroupBuild } from 'pnc-api-types-ts';
+import { useState, useEffect, useCallback } from 'react';
+import { DataContainer } from '../../containers/DataContainer/DataContainer';
+import { IService, useDataContainer } from '../../containers/DataContainer/useDataContainer';
+import { buildService } from '../../services/buildService';
+import { groupBuildService } from '../../services/groupBuildService';
+import { isBuild } from '../../utils/entityRecognition';
+import styles from './DependencyTree.module.css';
+import { BuildStatus } from '../BuildStatus/BuildStatus';
+
+interface IGraphEdge {
+  source: string;
+  target: string;
+}
+
+interface IGraphVertic {
+  name: string;
+  dataType: string;
+  data: Build | GroupBuild;
+  _dependencyBuildIds: Array<string>;
+  _dependentBuildIds: Array<string>;
+}
+
+interface IDependencyTreeProps {
+  build?: Build;
+  groupBuild?: GroupBuild;
+}
+
+interface IServiceDataProps {
+  build?: Build;
+  groupBuild?: GroupBuild;
+}
+
+interface IDependencyBuild extends Build, GroupBuild {
+  _buildIds?: Array<string>;
+  _dependencyBuildIds?: Array<string>;
+  _dependentBuildIds?: Array<string>;
+}
+
+interface IDependencyDataItem extends TreeViewDataItem {
+  data: IDependencyBuild;
+  children?: IDependencyDataItem[];
+  level: number;
+}
+
+/**
+ * Display a dependency tree for builds or group builds.
+ *
+ * @param build - The current build of the dependency tree
+ * @param groupBuild - The current group build of the dependency tree
+ *
+ */
+export const DependencyTree = ({ build, groupBuild }: IDependencyTreeProps) => {
+  const [visible, setVisible] = useState<boolean>(true);
+  const [dependentStructure, setDependentStructure] = useState<Array<IDependencyDataItem>>();
+  const [dependencyStructure, setDependencyStructure] = useState<IDependencyDataItem>();
+  const [buildItem, setBuildItem] = useState<IDependencyBuild>();
+  const [allExpanded, setAllExpanded] = useState<boolean | undefined>(undefined);
+  const dataContainer = useDataContainer(
+    useCallback(({ serviceData, requestConfig }: IService<IServiceDataProps>) => {
+      if (serviceData!.build) {
+        return buildService.getDependencyGraph({ id: serviceData!.build!.id }, requestConfig);
+      } else {
+        return groupBuildService.getDependencyGraph({ id: serviceData!.groupBuild!.id }, requestConfig);
+      }
+    }, [])
+  );
+
+  const refreshDataContainer = dataContainer.refresh;
+
+  const refreshComponent = () => {
```

Comment:
isn't possible to use just `useEffect` or some other way? This implementation looks like workaround a little bit

Reply:
So far there is no better solution since what we are doing here is to refresh the component using conditions. because patternfly do not provide a way to manipulate the expand status from outside after initialization.